base-template-symlink.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+# Base Git hook template from https://github.com/rycus86/githooks
+#
+# It allows you to have a .githooks folder per-project that contains
+# its hooks to execute on various Git triggers.
+#
+
+INSTALL_DIR=$(git config --global githooks.installDir)
+GITHOOKS_TEMPLATE="$INSTALL_DIR/release/base-template.sh"
+
+if [ ! -d "$INSTALL_DIR" ]; then
+    echo "! Githooks installation is corrupt! " >&2
+    echo "  Install directory at '${INSTALL_DIR}' is missing." >&2
+    echo "  Please run the Githooks install script again to fix it." >&2
+    exit 1
+elif [ ! -f "$GITHOOKS_TEMPLATE" ]; then
+    echo "! Githooks link to '$GITHOOKS_TEMPLATE' is broken!"
+    echo "  Please run the Githooks install script again to fix it." >&2
+    exit 1
+fi
+
+# shellcheck disable=SC1090
+sh "$GITHOOKS_TEMPLATE" "$0" "$INSTALL_DIR" "$@"
```

Comment:
Jeah, but the point was to try (at least) not to rely anymore on env variables, its kind of nasty, but your approach would work! What do you think?

Reply:
Hm... I thought of with shifting
```shell
HOOK_NAME=$(basename "$0")
HOOK_FOLDER=$(dirname "$0")
sh "$GITHOOKS_TEMPLATE" "$HOOK_NAME" "$HOOK_FOLDER" "$INSTALL_DIR" "$@"
```
and I think no tests work, because how can we know how much to shift leaving us in rewriting all `base-template.sh` calls in the tests (39 in total)
or 
```shell
HOOK_NAME=$(basename "$0")
HOOK_FOLDER=$(dirname "$0")
GITHOOKS_HOOK_NAME="$HOOK_NAME" GITHOOKS_INSTALL_DIR="$INSTALL_DIR" GITHOOKS_HOOK_FOLDER="$HOOK_FOLDER" sh "$GITHOOKS_TEMPLATE" "$@"
```
with no shifting and all tests work...

Hm...

Reply:
I'd rather would have proper arguments instead of env variables. Any good hooks manager does it like that I think -> also `git lfs post-merge ...`

Reply:
or we replace all calls to `base-template.sh` in the tests with `base-template-wrapper.sh` and do some trickery replacement in this file in the dockerfile... that might work

Reply:
hm... an env. approach might be the least trouble some...

Reply:
I'd vote for environment variables, I don't think they are too bad. Do you have anything in particular that you know about that makes them "nasty"?
I think we get some env variables from Git itself too, don't we?
In any case, I think using them might be the least amount of change for now, then we can come back to arguments vs. env variables later?

Reply:
What I don't like about env vars is that they have a global shared namespace and it is somewhat easy to accidentally read them from somewhere else, especially when you're migrating interfaces (e.g. if you add a new env var later, then any existing "symlink" scripts will not set it, but if the env var was already set elsewhere, you'd *think* the symlink script has set it even when it did not.

However, with commandline arguments, change might also be non-trivial, since if you add an argument, you'd have to somehow know that you did.

> In any case, I think using them might be the least amount of change for now, then we can come back to arguments vs. env variables later?

One thing to keep into account when changing is that you might need to support existing "symlink" scripts that still use the old interface. Or are all symlink scripts already updated on update? The idea is to not do that, right?

Reply:
Yeah symlink scripts should be as simple as we possible can! With my last commit, I tried that, also the input from @matthijskooijman is good I think. I could also get rid easly of the env variables. As all global variables (and sh itself too because we cannot use `local` variables) are a realy Pain in the Ass... for the reason I wont elaborate here tl;dr...
So rather use function, be stateless as far as we can be (ok, its shell script ;-) lool). Also we had a weird bug because some GIT_DIR env variable was leaking from git and so on inside our githook, there is still a note about this... Thats the sad other story about these kind of stupid variables also they are handy I see :-)